### DISCUSSION TITLE
V9 - custom config for ImageSharp?

### DISCUSSION THREAD

**Discussion by arknu** [umbraco/Umbraco-CMS#10840] [Features and ideas]
I know that Umbraco registers ImageSharp.Web in the AddUmbraco and UseUmbracoMethods, specifically the AddUmbracoImageSharp method. However, from what I can see, there is not way to specify any custom options, for example for some of the event handlers. Shouldn't this be possible?

**Comment by bergmania** [comment]
My guess would be you can do something like this after the `services.AddUmbraco...` in `Startup.cs`
```
services.PostConfigure<ImageSharpMiddlewareOptions>(options =>
{
    // Whatever you wanna change
    //options.BrowserMaxAge = TimeSpan.Zero;
});
```

**Reply by ronaldbarendse** [reply]
You can indeed use `PostConfigure` to set the ImageSharp.Web middleware options and after PR https://github.com/umbraco/Umbraco-CMS/pull/10623 is merged, you can also change the [ImageSharp configuration](https://docs.sixlabors.com/articles/imagesharp/configuration.html) used for all image manipulations (check the comments in this PR for more information).

**Comment by arknu** [comment]
Thank you @bergmania and @ronaldbarendse. I'm still learning ASP.NET Core, so hadn't come across that one yet.

**Reply by ronaldbarendse** [reply]
I'd recommend looking up the Microsoft documentation, especially the fundamentals about [configuration](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0) and [options](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-5.0). This contains a great amount of useful information and examples to get started!

Using `PostConfigure` will ensure it's always executed after all other `Configure` calls. In most cases it will already be the last one to run though, so I'd recommend only using `PostConfigure` in the consuming project/web application (because packages can also update configuration). And you can also do this via an `IComposer` instead of within the `ConfigureServices` method in `Startup.cs`, but that's an Umbraco specific thing 😉
